### PR TITLE
Add ifdef to remove BDF for large mechanisms on the GPU

### PR DESCRIPTION
**Pull request by jrood-nrel**

### CODE REVIEW

**Review by marchdf — APPROVED** [review]

File: Reactions/ReactorBDF.cpp [hunk 1]
```diff
@@ -139,6 +139,7 @@ ReactorBDF::react(
 #endif
 )
 {
+#if !defined(AMREX_USE_GPU) || (NUM_SPECIES < 50)
```
Comment (PaulMullowney) [inline Reactions/ReactorBDF.cpp]: dodecane_lu compiles so maybe make it 54?

**Comment by PaulMullowney** [comment]
Do we need to be a little more careful with this? Doesn't BDF build on Nvidia with large chemistry models?

**Comment by jrood-nrel** [comment]
I'm waiting for machines to come out of maintenance and then I will test this.

**Comment by hsitaram** [comment]
I had checked builds and runs on nvidia gpus but not with a super large chemistry. i wonder if doing GpuArray instead of static arrays will fix the problem. ReactorCvode does gpuarrays as well

**Comment by jrood-nrel** [comment]
This solves the issue for me. I'm going to merge this because it currently fixes something that is broken anyway.

**Pull request merged**